core/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintFactory.java

```diff
@@ -192,6 +192,17 @@ public interface ConstraintFactory {
      */
     <A> @NonNull BiConstraintStream<A, A> forEachUniquePair(@NonNull Class<A> sourceClass, @NonNull BiJoiner<A, A>... joiners);
 
+    // ************************************************************************
+    // staticData
+    //************************************************************************
+
+    /**
+     * Computes and caches the tuples that would be produced by the given stream.
+     * As this is cached, it is vital the stream does not reference any variables
+     * (genuine or otherwise).
+     */
+    <A> @NonNull UniConstraintStream<A> staticData(UniConstraintStream<A> stream);
```

Comment:
What happens when this stream is used both statically and non-statically?
(Think node-sharing.)

Reply:
The output stream can be node shared if the input streams are identical (not implemented currently since this is a POC). The input stream will not be node shared; its an independent network.